src/client/Client.cc

```diff
@@ -2362,6 +2362,18 @@ void Client::handle_client_session(const MConstRef<MClientSession>& m)
 
       session->mds_features = std::move(m->supported_features);
       session->mds_metric_flags = std::move(m->metric_spec.metric_flags);
+      cap_auths = std::move(m->cap_auths);
+
+      /*
+       * Just force the session to be readonly if root_sqaush is enabled
+       * and the MDS doesn't support CEPHFS_FEATURE_MDS_AUTH_CAPS_CHECK.
+       */
+      if (root_squash_in_cap_auths() &&
+          !session->mds_features.test(CEPHFS_FEATURE_MDS_AUTH_CAPS_CHECK)) {
+        ldout(cct, 10) << " close the session, root_squash enabled with old MDS"
+                       << dendl;
+        force_session_readonly(session.get());
+      }
```

Comment:
I'm puzzled by this. How would a newer client know `root_squash_in_cap_auths()` if an old MDS cannot tell it `cap_auths` in the session message?

Reply:
Yeah, a good question and I need to fix this.

Reply:
@batrick I will just remove this check for old `MDS`, IMO it's hard to fix this unless we dencrypt the cephx info in client side, this is easy to do in userspace client but not kclient.

Or any better idea ?